lnst/Controller/ContainerPoolManager.py

```diff
@@ -285,6 +299,31 @@ def _connect_to_network(self, container: "Container", network: "Network"):
             "network": network.name,
         }
 
+        logging.debug(f"Added {eth} with {interface['address']}")
+
+        if check_process_running("firewalld"):
+            # add interface to trusted zone
+            host_interfaces = loads(
+                subprocess.check_output(
+                    "ip -json link",
+                    shell=True
+                ).decode("utf-8")
+            )
+            matched_host_interfaces = [
+                host_iface for host_iface in host_interfaces \
+                    if host_iface["ifindex"] == interface["link_index"]
```

Comment:
if i understand it correctly, `interface` is defined at https://github.com/LNST-project/lnst/blob/master/lnst/Controller/ContainerPoolManager.py#L271 which is also a json output of an `ip` command

if that's true then i think `interface['ifname']` should also be defined, and so i think you don't have to run this search for the interface?

am i missing something?

Reply:
maybe `interface` is INSIDE the container and you're looking for the `ifname` OUTSIDE the container?

Reply:
Yeah, the problem is that `ip link` shows only ifindex of the device in the other namespace, not the device name.